src/system_descriptor.jl

```diff
@@ -1,3 +1,21 @@
+struct SystemArgument
+    name::Symbol
+    default::Any
+    allowed_values::Set{<:Any}
+
+    function SystemArgument(name, default, allowed_values)
+        isempty(allowed_values) && error("allowed_values cannot be empty")
+        new(name, default, allowed_values)
+    end
+end
```

Comment:
It seems like we have been miscommunicating. I liked the kwarg constructor you had. `Specifying `name = value` at the call sites in system_descriptor_data.jl was helpful. I wanted you to put the error check in the inner constructor _and_ keep the kwarg-only outer constructor.